### PR TITLE
yt/go/sdk: refactor YT_PROXY cluster URL handling

### PR DESCRIPTION
**Pull request by koct9i**
* handle https scheme in YT_PROXY
* fix several HTTPS/TVM/... cases
* add unit test with same test cases as python SDK

### CODE REVIEW

**Comment by koct9i** [comment]
For the record: YT-20028

File: yt/go/yt/config_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+			config{proxy: "localhost:23924"},
+			"http://localhost:23924",
+		},
+		{
+			"short name w port",
+			config{proxy: "hostname:123"},
+			"http://hostname:123",
+		},
+		{
+			"short_host_w_schame",
```
Comment (tinarsky) [inline yt/go/yt/config_test.go:42]: schame -> scheme

File: yt/go/yt/config_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
+package yt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestClusterUrl(t *testing.T) {
```
Comment (tinarsky) [inline yt/go/yt/config_test.go:10]: Url -> URL

File: yt/go/yt/config_test.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+			"ipv6 with scheme and port",
+			config{proxy: "https://[::1]:23924"},
+			"https://[::1]:23924",
+		},
+	} {
+		if test.config.defaultSuffix != "" {
+			continue // Not supported yet
+		}
+
+		t.Logf("checking %q", test.name)
```
Comment (tinarsky) [inline yt/go/yt/config_test.go:156]: I propose using t.Run instead of printing the test name.

**Review by tinarsky — CHANGES_REQUESTED** [review]

**Review by tinarsky — APPROVED** [review]
LGTM

**Pull request closed without merging**